management/src/test/java/org/ehcache/docs/ManagementTest.java

```diff
@@ -99,8 +99,10 @@ public void usingManagementRegistry() throws Exception {
 
         // hit count is a sampled stat, for example its values could be [0,0,3,4].
         // In the present case, only the last value is important to us , the cache was eventually hit 4 times
-        int mostRecentIndex = onHeapStore_Hit_Count.getValue().length - 1;
-        onHeapHitCount = onHeapStore_Hit_Count.getValue()[mostRecentIndex].getValue();
+        if (onHeapStore_Hit_Count.getValue().length > 0) {
```

Comment:
long val = -1;
      // it could be several seconds before the sampled stats could become available
      // let's try until we find the correct value : 2
      do {
        ResultSet<ContextualStatistics> counters = query.execute();

        ContextualStatistics statisticsContext1 = counters.getResult(context1);

        CounterHistory counterContext1 = statisticsContext1.getStatistic(CounterHistory.class, "Cache:MissCount");

        // miss count is a sampled stat, for example its values could be [0,1,2].
        // In the present case, only the last value is important to us , the cache was eventually missed 2 times
        int mostRecentSampleIndex = counterContext1.getValue().length - 1;
        if(mostRecentSampleIndex >= 0) {
          val = counterContext1.getValue()[mostRecentSampleIndex].getValue();
        }
      } while(val != 2);